src/test/java/uk/gov/dwp/dataworks/DataKeyServiceApplicationTests.java

```diff
@@ -1,15 +1,126 @@
 package uk.gov.dwp.dataworks;
 
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.junit.Before;
 import org.junit.Test;
 import org.junit.runner.RunWith;
+import org.mockito.Mockito;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
 import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.context.ActiveProfiles;
 import org.springframework.test.context.junit4.SpringRunner;
+import org.springframework.test.web.servlet.MockMvc;
+import uk.gov.dwp.dataworks.dto.DecryptDataKeyResponse;
+import uk.gov.dwp.dataworks.dto.GenerateDataKeyResponse;
+import uk.gov.dwp.dataworks.errors.CurrentKeyIdFailure;
+import uk.gov.dwp.dataworks.errors.DataKeyDecryptionFailure;
+import uk.gov.dwp.dataworks.errors.DataKeyGenerationFailure;
+import uk.gov.dwp.dataworks.errors.GarbledDataKeyError;
+import uk.gov.dwp.dataworks.provider.CurrentKeyIdProvider;
+import uk.gov.dwp.dataworks.provider.DataKeyDecryptionProvider;
+import uk.gov.dwp.dataworks.provider.DataKeyGeneratorProvider;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.BDDMockito.given;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.content;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
 
 @RunWith(SpringRunner.class)
-@SpringBootTest
+@SpringBootTest()
+@ActiveProfiles("Test")
+@AutoConfigureMockMvc
 public class DataKeyServiceApplicationTests {
 
+    @Autowired
+    private CurrentKeyIdProvider currentKeyIdProvider;
+
+    @Autowired
+    private DataKeyGeneratorProvider dataKeyGeneratorProvider;
+
+    @Autowired
+    private DataKeyDecryptionProvider dataKeyDecryptionProvider;
+
+    @Autowired
+    private MockMvc mockMvc;
+
+    @Before
+    public void setup() {
+        Mockito.reset(currentKeyIdProvider, dataKeyGeneratorProvider, dataKeyDecryptionProvider);
+    }
+
     @Test
     public void contextLoads() {
     }
+
+    @Test
+    public void pingWorks() throws Exception {
```

Comment:
The test names here are quite implementation orientated. Not sure if there is a test name convention or not, but may be worth looking at them being more functional. I.e something like "shouldSucceedWhenRequestingPing" or "shouldReturnValidKeyWhenRequestingKeyGeneration".